lib/ecto/adapters/sql.ex

```diff
@@ -245,6 +245,80 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  @doc """
+  Executes an EXPLAIN statement or similar for the given query according to its kind and the
+  adapter in the given repository.
+
+  ## Examples
+
+      # Postgres
+      iex> Ecto.Adapters.SQL.explain(:all, Repo, Post)
+      "Seq Scan on posts p0  (cost=0.00..12.12 rows=1 width=443)"
+
+      # MySQL
+      iex> Ecto.Adapters.SQL.explain(:all, from(p in Post, where: p.title == "title")) |> IO.puts()
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+      | id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra       |
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+      |  1 | SIMPLE      | p0    | NULL       | ALL  | NULL          | NULL | NULL    | NULL |    1 |    100.0 | Using where |
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+
+      # Shared opts
+      iex> Ecto.Adapters.SQL.explain(:all, Repo, Post, analyze: true, timeout: 20_000)
+      "Seq Scan on posts p0  (cost=0.00..11.70 rows=170 width=443) (actual time=0.013..0.013 rows=0 loops=1)\\nPlanning Time: 0.031 ms\\nExecution Time: 0.021 ms"
+
+  It's safe to execute it for updates and deletes, no data change will be commited:
+
+      iex> Ecto.Adapters.SQL.explain(:update_all, Repo, from(p in Post, update: [set: [title: "new title"]]))
+      "Update on posts p0  (cost=0.00..11.70 rows=170 width=449)\\n  ->  Seq Scan on posts p0  (cost=0.00..11.70 rows=170 width=449)"
+
+  ### Opts
+
+  Built-in adapters support passing `opts` to the EXPLAIN statement according to the following:
+
+  Adapter          | Supported opts                                                               | Notes
+  ---------------- | --------------                                                               | -----
```

Comment:
Please see if this fixes the docs:

```suggestion
  :--------------- | :-------------                                                               | :----
```

Reply:
Given this markdown:

```md
Adapter          | Supported opts | Notes
:--------------- | :------------- | :----
MyXQL            | None           | `EXTENDED` and `PARTITIONS`
```

It generates the correct HTML:

```html
<table><thead><tr><th style="text-align: left;">Adapter</th><th style="text-align: left;">Supported opts</th><th style="text-align: left;">Notes</th></tr></thead><tbody><tr><td style="text-align: left;">MyXQL</td><td style="text-align: left;">None</td><td style="text-align: left;"><code class="inline">PARTITIONS</code> and <code class="inline">EXTENDED</code></td></tr></tbody></table>
```

But given this:

```md
Adapter          | Supported opts | Notes
:--------------- | :------------- | :----
MyXQL            | None           | `EXTENDED` and `PARTITIONS` opts
```

Generates:

```html
<table><thead><tr><th style="text-align: left;">Adapter</th><th style="text-align: left;">Supported opts</th><th style="text-align: left;">Notes</th></tr></thead><tbody><tr><td style="text-align: left;">MyXQL</td><td style="text-align: left;">None</td><td style="text-align: left;"> opts<code class="inline">PARTITIONS</code> and <code class="inline">EXTENDED</code></td></tr></tbody></table>
```

For some reason the "opts" word in rendered before the inlined codes:

<img width="487" alt="Screen Shot 2020-06-07 at 9 45 52 AM" src="https://user-images.githubusercontent.com/36407/83970430-b18a7c00-a8a3-11ea-9a3b-3ab29633b2ca.png">

Using latest ex_doc, running `mix compile && ex_doc "ecto_sql", "1", _build/dev/lib/ecto_sql/ebin`

Reply:
Looks like a bug? I can open an issue or take a look on this.

Reply:
Sounds like a Earmark bug. Note you can generate the docs like this: `MIX_ENV=docs mix docs`. Then I would try updating the Earmark version and see if it fixes it!

Reply:
Same result using latest versions:

```
% cat mix.lock | grep ex_doc
  "ex_doc": {:hex, :ex_doc, "0.22.1", "9bb6d51508778193a4ea90fa16eac47f8b67934f33f8271d5e1edec2dc0eee4c", [:mix], [{:earmark, "~> 1.4.0", [hex: :earmark, repo: "hexpm", optional: false]}, {:makeup_elixir, "~> 0.14", [hex: :makeup_elixir, repo: "hexpm", optional: false]}], "hexpm", "d957de1b75cb9f78d3ee17820733dc4460114d8b1e11f7ee4fd6546e69b1db60"},
```

```
cat mix.lock | grep earmark
  "earmark": {:hex, :earmark, "1.4.5", "62ffd3bd7722fb7a7b1ecd2419ea0b458c356e7168c1f5d65caf09b4fbdd13c8", [:mix], [], "hexpm", "b7d0e6263d83dc27141a523467799a685965bf8b13b6743413f19a7079843f4f"},
  "ex_doc": {:hex, :ex_doc, "0.22.1", "9bb6d51508778193a4ea90fa16eac47f8b67934f33f8271d5e1edec2dc0eee4c", [:mix], [{:earmark, "~> 1.4.0", [hex: :earmark, repo: "hexpm", optional: false]}, {:makeup_elixir, "~> 0.14", [hex: :makeup_elixir, repo: "hexpm", optional: false]}], "hexpm", "d957de1b75cb9f78d3ee17820733dc4460114d8b1e11f7ee4fd6546e69b1db60"},
```

I found a related open issue https://github.com/pragdave/earmark/issues/335, so I'll just comment there.